src/pages/account/mijn-mappen/[folderSlug]/index.tsx

```diff
@@ -126,65 +120,78 @@ const AccountMyCollections: NextPage = () => {
 	 */
 
 	useEffect(() => {
-		if (!activeCollection && collections) {
-			const favorites = collections?.items.find((col) => col.isDefault);
-			!blockFallbackRedirect && favorites && router.push(createCollectionSlug(favorites));
+		if (!activeFolder && folders) {
+			const favorites = folders?.items.find((col) => col.isDefault);
+			!blockFallbackRedirect && favorites && router.push(createFolderSlug(favorites));
 		}
-	}, [activeCollection, collections, router, blockFallbackRedirect]);
+	}, [activeFolder, folders, router, blockFallbackRedirect]);
 
 	useEffect(() => {
-		if (activeCollection && folderMedia.isStale) {
+		if (activeFolder && folderMedia.isStale) {
 			folderMedia.refetch();
 		}
-	}, [activeCollection]); // eslint-disable-line react-hooks/exhaustive-deps
+	}, [activeFolder]); // eslint-disable-line react-hooks/exhaustive-deps
 
 	/**
 	 * Events
 	 */
 
-	const onCollectionTitleChanged = (collection: Folder) => {
+	const onFolderTitleChanged = async (newFolder: Folder) => {
 		setBlockFallbackRedirect(true);
 
-		getCollections.refetch().then(() => {
-			router.push(createCollectionSlug(collection)).then(() => {
-				setBlockFallbackRedirect(false);
+		// Temp set folders with new name in redux store
+		if (folders) {
+			const newFolders: Folder[] = (folders?.items || []).map((folder): Folder => {
+				if (folder.id === newFolder.id) {
+					return newFolder;
+				} else {
+					return folder;
+				}
 			});
-		});
+			dispatch(
+				setFolders({
+					...folders,
+					items: newFolders,
+				})
+			);
+		}
+
+		// Fetch folders from the network
+		await getFolders.refetch();
+		await router.push(createFolderSlug(newFolder));
+		setBlockFallbackRedirect(false);
 	};
```

Comment:
Avoids `newFolder` and instantiating a one-use variable.

```suggestion
	const onFolderTitleChanged = async (folder: Folder) => {
		setBlockFallbackRedirect(true);

		// Temp set folders with new name in redux store
		folders &&
			dispatch(
				setFolders({
					...folders,
					items: (folders.items || []).map(
						(old): Folder => (folder.id === old.id ? folder : old)
					),
				})
			);

		// Fetch folders from the network
		await getFolders.refetch();
		await router.push(createFolderSlug(folder));
		setBlockFallbackRedirect(false);
	};
```